cmd/sanitize-prow-jobs/main.go

```diff
@@ -97,28 +138,64 @@ func determinizeJobs(prowJobConfigDir string) error {
 	return utilerrors.NewAggregate(errs)
 }
 
-func defaultJobConfig(jc *prowconfig.JobConfig) {
+func defaultJobConfig(jc *prowconfig.JobConfig, path string, config *Config) {
 	for k := range jc.PresubmitsStatic {
 		for idx := range jc.PresubmitsStatic[k] {
-			defaultJobBase(&jc.PresubmitsStatic[k][idx].JobBase)
+			defaultJobBase(&jc.PresubmitsStatic[k][idx].JobBase, path, config)
 		}
 	}
 	for k := range jc.PostsubmitsStatic {
 		for idx := range jc.PostsubmitsStatic[k] {
-			defaultJobBase(&jc.PostsubmitsStatic[k][idx].JobBase)
+			defaultJobBase(&jc.PostsubmitsStatic[k][idx].JobBase, path, config)
 		}
 	}
 	for idx := range jc.Periodics {
-		defaultJobBase(&jc.Periodics[idx].JobBase)
+		defaultJobBase(&jc.Periodics[idx].JobBase, path, config)
 	}
 }
 
-func defaultJobBase(jb *prowconfig.JobBase) {
+func defaultJobBase(jb *prowconfig.JobBase, path string, config *Config) {
+	jb.Cluster = getCluster(jb.Name, path, config)
+}
 
-	if jb.Cluster == "" || jb.Cluster == "default" {
-		jb.Cluster = defaultCluster
+func getCluster(name string, path string, config *Config) string {
+	distribution := config.Clusters.Distribution
+	clusters := make([]string, 0, len(distribution))
+	for cluster := range distribution {
+		clusters = append(clusters, cluster)
 	}
+	sort.Strings(clusters)
 
+	//jobs override files which override groups
+	for _, cluster := range clusters {
+		for _, job := range distribution[cluster].Jobs {
+			if name == job {
+				return cluster
+			}
+		}
+	}
+	for _, cluster := range clusters {
+		for _, file := range distribution[cluster].Files {
+			re, ok := config.Clusters.REMap[file]
+			if !ok {
+				continue
```

Comment:
I think this should never happen and hence this is an error or not?

Reply:
Done.